packages/rich-text/src/formatters/gradientColor.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import {
+  createSolid,
+  generatePatternStyles,
+  getGradientStyleFromColor,
+  isPatternEqual,
+  getColorFromGradientStyle,
+  type Gradient,
+  type Pattern,
+  DEFAULT_GRADIENT,
+} from '@googleforcreators/patterns';
+import type { EditorState, DraftInlineStyle } from 'draft-js';
+import type { CSSProperties } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { NONE, MULTIPLE_VALUE, GRADIENT_COLOR } from '../customConstants';
+import {
+  togglePrefixStyle,
+  getPrefixStylesInSelection,
+} from '../styleManipulation';
+import { isStyle, getVariable } from './util';
+
+const styleToColor = (style: string): Gradient =>
+  getColorFromGradientStyle(getVariable(style, GRADIENT_COLOR));
+
+const colorToStyle = (color: Gradient): string =>
+  `${GRADIENT_COLOR}-${getGradientStyleFromColor(color)}`;
+
+function elementToStyle(element: HTMLElement): string | null {
+  const isSpan = element.tagName.toLowerCase() === 'span';
+  const rawBackground = element.style.backgroundImage;
+  const hasBackground = Boolean(rawBackground);
+
+  if (isSpan && hasBackground) {
+    const gradient = getColorFromGradientStyle(rawBackground);
+    return colorToStyle(gradient);
+  }
+
+  return null;
+}
+
+function stylesToCSS(styles: DraftInlineStyle): null | CSSProperties {
+  const colorStyle = styles.find((someStyle) =>
+    isStyle(someStyle, GRADIENT_COLOR)
+  );
+
+  if (!colorStyle) {
+    return null;
+  }
+
+  let color: Pattern;
+  try {
+    color = styleToColor(colorStyle);
+  } catch (e) {
+    return null;
+  }
+
+  const { backgroundImage } = generatePatternStyles(color);
+
+  return {
+    backgroundImage,
+    backgroundClip: 'text',
+    WebkitBackgroundClip: 'text',
+    WebkitTextFillColor: 'transparent',
```

Comment:
I think the `-webkit-` prefixed ones should to come first, otherwise the override the unprefixed one.

<img width="396" alt="Screenshot 2024-03-19 at 10 21 36" src="https://github.com/GoogleForCreators/web-stories-wp/assets/841956/2193969e-3756-4603-9dc3-a3ea0ef4ef6b">

Reply:
Done!